dfget/locator/static_locator.go

```diff
@@ -0,0 +1,155 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package locator
+
+import (
+	"math/rand"
+	"sync/atomic"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/dfget/config"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+	"github.com/dragonflyoss/Dragonfly/pkg/netutils"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+const staticLocatorGroupName = "config"
+
+// StaticLocator uses the nodes passed from configuration or CLI.
+type StaticLocator struct {
+	idx   int32
+	Group *SupernodeGroup
+}
+
+// ----------------------------------------------------------------------------
+// constructors
+
+// NewStaticLocator constructs StaticLocator which uses the nodes passed from
+// configuration or CLI.
+func NewStaticLocator(nodes []*config.NodeWeight) *StaticLocator {
+	locator := &StaticLocator{}
+	if nodes == nil || len(nodes) == 0 {
+		return locator
+	}
+	group := &SupernodeGroup{
+		Name: staticLocatorGroupName,
+	}
+	for _, node := range nodes {
+		ip, port := netutils.GetIPAndPortFromNode(node.Node, config.DefaultSupernodePort)
+		if ip == "" {
+			continue
+		}
+		supernode := &Supernode{
+			Schema:    config.DefaultSupernodeSchema,
+			IP:        ip,
+			Port:      port,
+			Weight:    node.Weight,
+			GroupName: staticLocatorGroupName,
+		}
+		for i := 0; i < supernode.Weight; i++ {
+			group.Nodes = append(group.Nodes, supernode)
+		}
+	}
+	shuffleNodes(group.Nodes)
+	locator.Group = group
+	return locator
+}
+
+// NewStaticLocatorFromStr constructs StaticLocator from string list.
+// The format of nodes is: ip:port=weight
+func NewStaticLocatorFromStr(nodes []string) (*StaticLocator, error) {
+	nodeWeight, err := config.ParseNodesSlice(nodes)
+	if err != nil {
+		return nil, err
+	}
+	return NewStaticLocator(nodeWeight), nil
+}
+
+// ----------------------------------------------------------------------------
+// implement api methods
+
+func (s *StaticLocator) Get() *Supernode {
+	if s.Group == nil {
+		return nil
+	}
+	idx := s.load()
+	if idx >= len(s.Group.Nodes) {
+		return nil
+	}
+	return s.Group.Nodes[idx]
+}
+
+func (s *StaticLocator) Next() *Supernode {
+	if s.Group == nil || s.load() >= len(s.Group.Nodes) {
+		return nil
+	}
+	idx := s.inc()
+	if idx >= len(s.Group.Nodes) {
+		return nil
+	}
+	return s.Group.Nodes[idx]
+}
+
+func (s *StaticLocator) GetGroup(name string) *SupernodeGroup {
+	if s.Group == nil || s.Group.Name != name {
+		return nil
+	}
+	return s.Group
+}
+
+func (s *StaticLocator) All() []*SupernodeGroup {
+	if s.Group == nil {
+		return nil
+	}
+	return []*SupernodeGroup{s.Group}
+}
+
+func (s *StaticLocator) Report(node string, metrics *SupernodeMetrics) {
+	// unnecessary to implement this method
+	return
+}
+
+func (s *StaticLocator) Refresh() bool {
```

Comment:
This looks more like Reset :sweat_smile: How about rename as `Reset`.

Reply:
This implementation looks more like `Reset`. But considering the different behaviors of other implementations, it may get a new list of supernodes via invoking this method.
So IMOP, `Reresh` is  more appropriate. WDYT?

Reply:
Make sense!